### ISSUE TITLE
Docker image - root and datadir

### ISSUE THREAD

**Issue by tkstanczak** [NethermindEth/nethermind#2139]
AlexSSD7Today at 18:30
ok, got it
I see your docker image is executed as root

which is not good
anyways, I tried to run with -u flag (which specifies the user)
and it throws permission error because it cannot access the /nethermind directory
is there a way to not use /nethermind directory?
I have another docker volume where I want everything to be stored
is it accomplishable?
tkstanczakToday at 18:33
@matilote willbe able to look at this - I think we had this conversation in the past and I would like to hear from him tomorrow if it would cause a lot of trouble to run it as a user
you can change the directory by configuring volumes mappings
in your Dockerfile / Docker compose
AlexSSD7Today at 18:34
I don't use neither dockerfiles nor docker compose
tkstanczakToday at 18:34
https://docs.nethermind.io/nethermind/ethereum-client/dockers
Dockers
Nethermind Dockers instructions

Volumes
To fully override a configuration file, you need to use a volume:
-v /home/user/mainnet.cfg:/nethermind/configs/mainnet.cfg
In order to map an internal database to a local volume, you need to map /nethermind/nethermind_db/:
-v /home/user/nethermind_db:/nethermind/nethermind_db
Logs can be mapped with the following command:
-v /home/user/logs:/nethermind/logs
Keystore:
-v /home/user/keystore:/nethermind/keystore
Nlog:
-v /home/user/NLog.config:/nethermind/NLog.config
More detailed NLog configuration can be found here.
AlexSSD7Today at 18:35
I have a docker volume, and it is possible to mount it to only one directory
everything is good unless it tries to access /nethermind/Nethermind directory
with geth I can just specify --datadir
and everything will be placed right to that directory
is it possible to do it with the nethermind?

**Comment by AlexSSD7** [comment]
To improve the Nethermind DevOps experience, the docker image should be more portable and friendly for custom user permissions. Right now the docker image is being executed by root, which is not a good practice.

For example, I have one docker volume, and I want everything to be placed there.

With other ethereum clients, I can just do this:
```sh
# Geth
docker run -v my-volume:/ethdb ethereum/client-go --datadir /ethdb

# OpenEthereum
docker run -v my-volume:/ethdb openethereum/openethereum --base-path /ethdb
```

Trying to run

```
docker run -v my-volume:/ethdb nethermind/nethermind --baseDbPath /ethdb
```

fails because nethermind tries to access `/nethermind/Nethermind` directory, and custom user doesn't has any access to it.

**Comment by matilote** [comment]
@AlexSSD7 I am aware of this non-root practice and it certainly might be helpful in some cases, however non-root containers are usually causing troubles when it comes their usage. You can always apply some extra security layers to your containers by using custom AppArmor profiles, dropping capabilities `--cap-drop ALL`  or you can use linux namespaces and user re-mapping (https://docs.docker.com/engine/security/userns-remap/).

As for the volume mapping, please try using this command instead:
```
docker run my-volume:/ethdb/nethermind_db nethermind/nethermind --baseDbPath /ethdb
```

**Comment by AlexSSD7** [comment]
Running non-system applications as root is very dangerous and is considered as a really bad practice.

**Comment by AlexSSD7** [comment]
Improving production grade deployment experience will help Nethermind to become more used.

**Comment by matilote** [comment]
I will try removing root from Dockerfile and see how it affects usage experience

Wysłano z usługi Yahoo Poczta w systemie Android 
 
  Dnia czw., lip 16, 2020 o 11:05, Alexander Sadovskyi<notifications@github.com> napisał:   


Improving production grade deployment experience will help Nethermind to become more popular.

**Comment by matilote** [comment]
Fixed with #2580

**Comment by matilote** [comment]
non-dockers -> https://docs.nethermind.io/nethermind/ethereum-client/configuration#datadir

dockers -> https://docs.nethermind.io/nethermind/ethereum-client/dockers#datadir
